design-proposals/sriov-interfaces-mapping.md

```diff
@@ -0,0 +1,760 @@
+
+The following sections go into more details.
+
+## Utilizing `k8s.v1.cni.cncf.io/network-status` annotation
+This annotation includes the information required to do one-to-one mapping between a VM interface and VF PCI address:
+
+### Example
+```yaml
+k8s.v1.cni.cncf.io/network-status: |-
+    [
+        {
+            "name": "kindnet",
+            "interface": "eth0",
+            "ips": [
+                "10.244.2.131"
+            ],
+            "mac": "82:cf:7c:98:43:7e",
+            "default": true,
+            "dns": {}
+        },{
+            "name": "default/sriov-network-vlan100",
+            "interface": "net1",
+            "dns": {},
+            "device-info": {
+                "type": "pci",
+                "version": "1.0.0",
+                "pci": {
+                    "pci-address": "0000:04:02.5"
+                }
+            }
+        },{
+            "name": "default/sriov-network-vlan200",
+            "interface": "net2",
+            "dns": {},
+            "device-info": {
+                "type": "pci",
+                "version": "1.0.0",
+                "pci": {
+                    "pci-address": "0000:04:02.2"
+                }
+            }
+        }
+    ]
+```
+
+> **_Note_**: Multus provides PCI device information since version [v3.7](https://github.com/k8snetworkplumbingwg/multus-cni/releases/tag/v3.7).<br/>
+> In order to maintain backward compatibility, when the annotation is missing, virt-launcher will fall to the current mapping method based on environment variables.<br/>
+> Once a Pod reaches Running state, the CNI operation is finished successfully and the annotation will present on the Pod.
+
+### Annotations Exposure Method
+A subset of the information in the `network-status` annotation needs to be passed down to the virt-launcher for composing the right domain configuration.
+
+There have been several methods explored, recorded in [Appendix 1: Alternative Annotations Exposure Methods](#alternative-solutions).
+
+In this section the chosen method is explained.
+
+#### Expose `network-status` annotation to virt-launcher Pod
+The annotation content can be exposed to virt-launcher application using Kubernetes Downward API [[1]](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/):
+
+virt-controller creates an annotation that will hold the mapping between the (VMI) SR-IOV networks and PCI addresses, e.g: `kubevirt.io/network-to-pci-address`.
+
+1. Given the `network-status` annotation, virt-controller process the content and compose 1:1 mapping between a (VMI) network from it and the device PCI address information.
+   Create `network-to-pci-address` annotation with the mapping information, for example:
+    ```yaml
+    kubevirt.io/network-to-pci-address: |-
```

Comment:
This is now focused on sriov solely. So good enough for me to fix the bug at hand. As discussed. :+1: 

Please add that on follow up changes there is still the expectation that you fall in line with the VMI status and remove this information flow and move it over to virt-handler.

This may live long, since temporary solutions tend to live long, but when you start to add more data I would expect that you follow the main communicatino paths and remove this workaround.

Reply:
>Please add that on follow up changes there is still the expectation that you fall in line with the VMI status and remove this information flow and move it over to virt-handler.

Do you mean to add it as comment to the commit or on the code?

Reply:
> Do you mean to add it as comment to the commit or on the code?

a short fixme or so would be nice too yes, but I meant in the proposal, so that we don't start future discussions at 0. Just that this is captured. Also the pros/cons sections are probably not necessarily correctly reflecting the latest discussion points, however I would not insist on reworking them. Just make it clear that eventually it has to be moved to vmi-status.